stable/spark/templates/spark-master-deployment.yaml

```diff
@@ -0,0 +1,134 @@
+  labels:
+    heritage: {{ .Release.Service | quote }}
+    release: {{ .Release.Name | quote }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    component: "{{ .Release.Name }}-{{ .Values.master.component }}"
+spec:
+  type: {{ .Values.webUi.serviceType }}
+  ports:
+    - port: {{ .Values.webUi.servicePort }}
+      nodePort: {{ .Values.webUi.nodePort }}
+      targetPort: {{ .Values.webUi.containerPort }}
+  externalIPs:
+    - "{{ .Values.externalIP.ip }}"
+  selector:
+    component: "{{ .Release.Name }}-{{ .Values.master.component }}"
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ .Values.master.name }}
+  labels:
+    heritage: {{ .Release.Service | quote }}
+    release: {{ .Release.Name | quote }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    component: "{{ .Release.Name }}-{{ .Values.master.component }}"
+    app: spark-master
+spec:
+  #podAntiAffinity:
+  #  requiredDuringSchedulingIgnoredDuringExecution:
+  #  - labelSelector:
+  #      matchExpressions:
+  #      - key: app
+  #        operator: In
+  #        values:
+  #        - spark-master
+  #    topologyKey: kubernetes.io/hostname
+  replicas: {{ default 1 .Values.master.replicas }}
+  strategy:
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      component: "{{ .Release.Name }}-{{ .Values.master.component }}"
+  template:
+    metadata:
+      labels:
+        heritage: {{ .Release.Service | quote }}
+        release: {{ .Release.Name | quote }}
+        chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+        component: "{{ .Release.Name }}-{{ .Values.master.component }}"
+    spec:
+      containers:
+        - name: {{ .Values.master.name }}
+          image: "{{ .Values.master.image }}:{{ .Values.master.imageTag }}"
+          args: ["master"]
+          ports:
+            - containerPort: {{ .Values.master.containerPort }}
+            - containerPort: {{ .Values.webUi.containerPort }}
+          resources:
+            requests:
+              cpu: "{{ .Values.master.cpu }}"
+              memory: "{{ .Values.master.memory }}"
+          env:
+          - name: SPARK_DAEMON_MEMORY
+            value: {{ default "1g" .Values.master.daemonMemory | quote }}
+          - name: SPARK_MASTER_HOST
+            value: master
+          - name: SPARK_MASTER_PORT
+            value: {{ .Values.master.servicePort | quote }}
+          - name: SPARK_MASTER_WEBUI_PORT
+            value: {{ .Values.webUi.containerPort | quote }}
+        - name: jupyter
+          imagePullPolicy: Always
+          command: ["start.sh"]
+          args: ["/usr/local/spark/bin/pyspark"]
+          env:
+            - name: PYSPARK_PYTHON
+              value: python3
+            - name: PYSPARK_DRIVER_PYTHON
+              value: ipython
+            - name: PYSPARK_DRIVER_PYTHON_OPTS
+              value: "notebook --NotebookApp.token={{ .Values.master.jupyter.token }}"
+          image: "cloudpg/spark-notebook:k8s-py3.6" 
```

Comment:
let user specify image and tag. Look at [here](https://github.com/DODAS-TS/helm_charts/blob/master/stable/cachingondemand/templates/cache-server-pod.yml#L88) for reference